smart_open/smart_open_lib.py

```diff
@@ -387,6 +387,34 @@ def _patch_pathlib(func):
     return old_impl
 
 
+def smart_open(
+        uri,
+        mode='rb',
+        buffering=-1,
+        encoding=None,
+        errors=None,
+        newline=None,
+        closefd=True,
+        opener=None,
+        ignore_ext=False,
+        **kwargs,
+        ):
+    #
+    # This is a thin wrapper of smart_open.open.  It's here for backward
+    # compatibility.  It works exactly like smart_open.open when the passed
+    # parameters are identical.  Otherwise, it raises a DeprecationWarning.
+    #
+    if kwargs:
+        raise DeprecationWarning(
+            'The following keyword parameters are not supported: %r. '
+            'See https://github.com/RaRe-Technologies/smart_open/blob'
+            '/develop/MIGRATING_FROM_OLDER_VERSIONS.rst '
+            'for more information.' % sorted(kwargs)
+        )
+    del kwargs
+    return open(**locals())
```

Comment:
If the actual end goal is true deprecation/removal of the function itself - not just some sets of supported parameters – perhaps OK to fail in this case, too. (Or at least: show some deprecation warning before proceeding?) However, if it's true that in this case, the caller can just use `smart_open.open()` with exactly the same params instead, the message could assure them that.

Reply:
We can't fail in this case because the gensim-data stuff will break.

I agree with the other suggestion.